### PR TITLE
GPC-NONE: Fix spring security deprecations

### PR DESCRIPTION
**Pull request by ethanmills**
Please review carefully. Per https://docs.spring.io/spring-security/reference/servlet/configuration/java.html#_multiple_httpsecurity_instances (kotlin docs just seem to be wrong)

### CODE REVIEW

File: src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt [hunk 1]
```diff
@@ -30,26 +31,48 @@ class ResourceServerConfiguration {
       .roles(role)
   }
 
+  @Order(1)
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt:34]: I got part way through this, could we make sure we've got a test for it too?

Comment (olevett) [reply src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt:34]: As a really quick starter, I think something like this:
```kotlin
package uk.gov.gdx.datashare.uk.gov.gdx.datashare.integration.health

import org.junit.jupiter.api.Test
import uk.gov.gdx.datashare.integration.SqsIntegrationTestBase

class PrometheusTest : SqsIntegrationTestBase() {
  @Test
  fun `Unauthenticated page is inaccessible`() {
    webTestClient.get()
      .uri("/prometheus")
      .exchange()
      .expectStatus()
      .isUnauthorized
  }

  @Test
  fun `Authenticated page is accessible`() {
    webTestClient
      .get()
      .uri("/prometheus")
      .headers { it.setBasicAuth("prometheus", "prometheus") }
      .exchange()
      .expectStatus()
      .isOk
  }
}
```

**Review by oskarwilliams — COMMENTED** [review]
If we can add some tests that would be great, but it all looks accurate to me.

File: src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt [hunk 2]
```diff
@@ -30,26 +31,48 @@ class ResourceServerConfiguration {
...
+      .securityMatcher("/prometheus")
+      .authorizeHttpRequests {
+        it.anyRequest().hasRole("ACTUATOR")
+      }
+      .httpBasic {}
+      .build()
+  }
+
+  @Bean
+  fun apiFilterChain(http: HttpSecurity): SecurityFilterChain {
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt:69]: Does no Order on this matter?

File: src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt [hunk 3]
```diff
@@ -30,26 +31,48 @@ class ResourceServerConfiguration {
...
+  }
+
+  @Order(2)
+  @Bean
+  fun prometheusFilterChain(http: HttpSecurity): SecurityFilterChain {
+    return http
+      .csrf { it.disable() }
+      .securityMatcher("/prometheus")
+      .authorizeHttpRequests {
+        it.anyRequest().hasRole("ACTUATOR")
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt:62]: Whilst we're here, we hard code the role on this line, but the role is configurable for the user itself - I wonder if we should either configure this one, or hard code the other role?

File: src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt (hunk 2 above)
Comment (ethanmills) [reply src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt:69]: Was just copied from the docs - happy to add LAST

File: src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt (hunk 3 above)
Comment (ethanmills) [reply src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt:62]: Yeah wasn't sure what that was about - @oskarwilliams  what was the intention?

**Review by oskarwilliams — APPROVED** [review]

File: src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt (hunk 2 above)
Comment (ethanmills) [reply src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt:69]: It's `Ordered.LOWEST_PRECEDENCE` which is `Integer.MAX_VALUE` under the hood 😄

File: src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt (hunk 3 above)
Comment (ethanmills) [reply src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt:62]: I'm looking to fiddle with the actuator stuff a bit, but will do that in a separate change. Will leave this until then

**Review by olevett — APPROVED** [review]

**Pull request merged**